apps/OpenSpace/ext/launcher/include/sgctedit/display.h

```diff
@@ -0,0 +1,90 @@
+ * Copyright (c) 2014-2022                                                               *
+ *                                                                                       *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy of this  *
+ * software and associated documentation files (the "Software"), to deal in the Software *
+ * without restriction, including without limitation the rights to use, copy, modify,    *
+ * merge, publish, distribute, sublicense, and/or sell copies of the Software, and to    *
+ * permit persons to whom the Software is furnished to do so, subject to the following   *
+ * conditions:                                                                           *
+ *                                                                                       *
+ * The above copyright notice and this permission notice shall be included in all copies *
+ * or substantial portions of the Software.                                              *
+ *                                                                                       *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED,   *
+ * INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS FOR A         *
+ * PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT    *
+ * HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF  *
+ * CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE  *
+ * OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.                                         *
+ ****************************************************************************************/
+
+#ifndef __OPENSPACE_UI_LAUNCHER___DISPLAY___H__
+#define __OPENSPACE_UI_LAUNCHER___DISPLAY___H__
+
+#include <QCheckBox>
+#include <QComboBox>
+#include <QIntValidator>
+#include <QLabel>
+#include <QLayout>
+#include <QLineEdit>
+#include <QPainter>
+#include <QPainterPath>
+#include <QPoint>
+#include <QPushButton>
+#include <QTextBrowser>
+#include <QVector>
+#include <QWidget>
+
+#include <vector>
+#include <iostream>
+
+#include "windowcontrol.h"
+#include "monitorbox.h"
+
+
+class Display : public QWidget
+{
+Q_OBJECT
+public:
+    explicit Display(MonitorBox* monitorRenderBox, std::vector<QRect>& monitorSizeList,
+        const unsigned int nMaxWindows, const QString* winColors);
+    ~Display();
+    std::vector<WindowControl*> windowControls();
+    unsigned int nWindows() const;
+    void uncheckWebGuiOptions();
+    void setindowChangeCallback(std::function<void(int, int, const QRectF&)> cb);
+
+private slots:
+    void addWindow();
+    void removeWindow();
+
+private:
+    void initializeWindowControl();
+    void removeWindowControl();
+    void initializeLayout();
+    void showWindows();
+    void addDisplayLayout(unsigned int column, MonitorBox* monBox, QVBoxLayout* layout);
+    MonitorBox* _monBox;
```

Comment:
Agreed with a smart pointer being a better choice. I switched to a `shared_ptr` since this needs to be used in other classes.
It's easier to work with raw pointers when using Qt however. A lot of examples online just use raw pointers and let Qt clean up the memory. It's not just memory management; I had to use the smart pointer's `get` when passing it to Qt, which then makes it an "in the wild" pointer again, but Qt should delete it so it should be OK. I don't know of another way to pass it to Qt.